nntrainer/layers/mol_attention_layer.h

```diff
@@ -16,14 +16,15 @@
 #ifdef __cplusplus
 
 #include <attention_layer.h>
+#include <layer_impl.h>
 
 namespace nntrainer {
 
 /**
  * @class   MoL Attention Layer
  * @brief   Mixture of Logistics Attention Layer
  */
-class MoLAttentionLayer : public AttentionLayer {
+class MoLAttentionLayer : public AttentionLayer, public LayerImpl {
```

Comment:
There seems no codebits in this class using AttentionLayer methods, could you explain why trying to enable multiple inheritance?

I think we shouldn't enable multiple inheritance for non-interface classes unless we absolutely need to

Reply:
multiple inheritance will be used once common code bits are removed from mol attention layer code and reused from attention layer.

Reply:
In that case, we can favor composition, or higher order class. My concern is making LayerImpl virtual, where we are opening a hatch all kinds of multiple inheritance will be possible.

Reply:
eg1) below will not work without try catch.

```
 AttentionLayer::setProperty(remain_props);
 LayerImpl::setProperty(remain_props);
 exporter.saveResult(mol_props, method, this);
```

eg2) attention layer(or other potential) starts to inherit LayerImpl, this will be wrong

```
 AttentionLayer::exportTo(exporter, method);
 LayerImpl::exportTo(exporter, method);
 exporter.saveResult(mol_props, method, this);
```

Reply:
Agreed, dual inheritance is creating issues with setProperty as pointed in the first example but IMO, LayerImpl should not throw on remaining properties but let the caller throw.
But this is beyond the scope of this PR, so removed inheritance to attention layer.

Reply:
Thanks :)